src/session.cpp

```diff
@@ -167,12 +169,16 @@ session_t::session_t(std::unique_ptr<logging::log_t> log_,
 void
 session_t::handle(const decoder_t::message_type& message) {
     const channel_map_t::key_type channel_id = message.span();
-
+    tracer::trace_restore_scope_t trace_scope;
+    if(message.trace_id() != 0) {
+        trace_scope.restore("sr", prototype->name(), message.trace_id(), message.span_id(), message.parent_id());
```

Comment:
Of course a convenient getters are required.

I mean headers are required for non-breaking protocol extension. Its representation may be binary, msgpack, protobuf or whatever else, but `decoded_message_t` users shouldn't care about it.